cmd/routed-eni-cni-plugin/driver/driver.go

```diff
@@ -677,3 +691,44 @@ func buildVlanLink(vlanName string, vlanID int, parentIfIndex int, eniMAC string
 	la.HardwareAddr, _ = net.ParseMAC(eniMAC)
 	return &netlink.Vlan{LinkAttrs: la, VlanId: vlanID}
 }
+
+type MACGenerator struct {
+	netlink   netlinkwrapper.NetLink
+	randMACfn func() string
+}
+
+func NewMACGenerator() MACGenerator {
+	return MACGenerator{netlink: netlinkwrapper.NewNetLink(), randMACfn: generateRandomMAC}
+}
+
+func generateRandomMAC() string {
+	mac := make([]byte, 6)
+	rand.Read(mac)
+	// Set the local bit and unset the multicast bit
+	mac[0] = (mac[0] | 2) & 0xfe
+
+	return fmt.Sprintf("%02x:%02x:%02x:%02x:%02x:%02x",
+		mac[0], mac[1], mac[2], mac[3], mac[4], mac[5])
+}
+
+// generateUniqueRandomMAC will compare randomly generated Mac to mac addresses of veth already present in host.
+func (m MACGenerator) generateUniqueRandomMAC() (string, error) {
+	ll, err := m.netlink.LinkList()
```

Comment:
We can't cache this ? I'm thinking about latency since we will be making a list call with every pod creation.

Reply:
had discussion internally. Can be improved if there are issues. Limitation with caching is we cannot track all link device on node.